From 158c80e46369fcd091cc763707432808b38eea89 Mon Sep 17 00:00:00 2001
From: Mugunthan V N <mugunthanvnm@ti.com>
Date: Fri, 18 May 2012 09:20:44 +0530
Subject: [PATCH 1023/1092] ARM: OMAP: AM33XX: NET: cpsw: add dual emac vlan
 id in platform data

Assigning vlan id 2 and 3 to dual EMACs respectively in CPSW platofrm data

Signed-off-by: Mugunthan V N <mugunthanvnm@ti.com>
---
 arch/arm/mach-omap2/devices.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-omap2/devices.c b/arch/arm/mach-omap2/devices.c
index f0b4bf1..1372224 100644
--- a/arch/arm/mach-omap2/devices.c
+++ b/arch/arm/mach-omap2/devices.c
@@ -1191,6 +1191,9 @@ static int __init omap2_init_devices(void)
 arch_initcall(omap2_init_devices);
 
 #define AM33XX_EMAC_MDIO_FREQ		(1000000)
+/* Port Vlan IDs for Dual Mac Mode */
+#define CPSW_PORT_VLAN_SLAVE_0		2
+#define CPSW_PORT_VLAN_SLAVE_1		3
 
 static u64 am33xx_cpsw_dmamask = DMA_BIT_MASK(32);
 /* TODO : Verify the offsets */
@@ -1199,11 +1202,13 @@ static struct cpsw_slave_data am33xx_cpsw_slaves[] = {
 		.slave_reg_ofs  = 0x208,
 		.sliver_reg_ofs = 0xd80,
 		.phy_id		= "0:00",
+		.dual_emac_reserved_vlan = CPSW_PORT_VLAN_SLAVE_0,
 	},
 	{
 		.slave_reg_ofs  = 0x308,
 		.sliver_reg_ofs = 0xdc0,
 		.phy_id		= "0:01",
+		.dual_emac_reserved_vlan = CPSW_PORT_VLAN_SLAVE_1,
 	},
 };
 
-- 
1.7.11.2

